core/client/README.md

```diff
@@ -0,0 +1,364 @@
+may specify an option as follows:
+
+```go
+// When broadcasting a transaction, wait until it is included in a block.
+txOpts := []ctypes.TxOpt{ctypes.WithSyncBroadcast(true)}
+txHash, err := cl.DeployDatabase(ctx, schema, txOpts...) // wait
+```
+
+In addition to broadcasting the transaction, this ensures the transaction was
+included in a block. The next section describes how to check the outcome of a
+transaction's *execution*.
+
+### Transaction Status
+
+After being broadcasted to a Kwil node, a transaction must be included in a
+block *and* execute without error for the database to actually be deployed. Use
+the `TxQuery` method to check.
+
+In our example app, we can define the following closure to use after every
+transaction we broadcast:
+
+```go
+// After broadcast, we get a transaction hash that uniquely identifies the
+// transaction. Use the TxQuery method to verify the execution succeeded.
+checkTx := func(txHash []byte, desc string) {
+	res, err := cl.TxQuery(ctx, txHash)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if res.TxResult.Code == transactions.CodeOk.Uint32() {
+		fmt.Printf("Success: %q in transaction %x\n", desc, txHash)
+	} else {
+		log.Fatalf("Fail: %q in transaction %x, Result code %d, log: %q",
+			desc, txHash, res.TxResult.Code, res.TxResult.Log)
+	}
+}
+
+txHash, err := cl.DeployDatabase(ctx, schema, txOpts...)
+if err != nil {
+	log.Fatal(err)
+}
+checkTx(txHash, "deploy database")
+```
+
+Combined the `WithSyncBroadcast` option in `txOpts`, this use of `TxQuery` will
+ensure the transaction executed without error, otherwise the application will exit.
+
+### Dropping a Database
+
+With a successfully deployed database, use the `DropDatabase` method to delete a database:
+
+```go
+txHash, err = cl.DropDatabase(ctx, dbName, txOpts...)
+if err != nil {
+	log.Fatal(err)
+}
+checkTx(txHash, "drop database")
+```
+
+NOTE: This is only permitted if you are the `owner` of the database i.e. you deployed it.
+
+### Action Execution
+
+As with the database deploy and drop methods, action *execution* requires a
+transaction since it is used to modify data in a schema.  Use the
+`ExecuteAction` method.
+
+In the schema deployed by our example app, the action called `"tag"` will insert data:
+
+```go
+const actionName = "tag"
+args := [][]any{{"jon was here"}} // one execution, one argument
+txHash, err := cl.ExecuteAction(ctx, dbid, actionName, args, txOpts...)
+if err != nil {
+	log.Fatal(err)
+}
+checkTx(txHash, "execute action")
+```
+
+In the above example, the schema's `"tag"` action is defined as:
+
+```js
+action tag($msg) public {
+    INSERT INTO "tags" (ident, msg) VALUES (@caller, $msg);
+}
+```
+
+We called the `"tag"` action with arguments `[][]any{{"jon was here"}}`. This is
+a `[][]any` to support batched action execution. In this example, we execute the
+action once, using `"jon was here"` as the `$msg` argument.
+
+NOTE: To execute an action with no input arguments, provide `nil`.
```

Comment:
> batched action execution

could you also provide an example of this `batched action execution`?

> using `"jon was here"` as the `$msg` argument.

So, I assume, that if there were more than one argument, would it always be in order? i.e.:
```
action tag($msg,$msg1)
```
will have
```
args := [][]any{{"jon was here"},{"ben was here"}}
```
where
msg1 -> `jon was here`
msg2 -> `ben was here`

Reply:
Oh, yes.  I had read the example incorrectly.  This is the revision to clarify: https://github.com/kwilteam/kwil-db/pull/662/commits/02577bde53868d8dd9293b083eb5b9ab71cabbcd#diff-037cd674919f54a98f50e6ee053384586c751a823ab95cb38e7cf948b125301cR326

Reply:
thanks!